cmd/gosky/debug.go

```diff
@@ -244,3 +251,76 @@ var debugStreamCmd = &cli.Command{
 		return nil
 	},
 }
+
+var debugFeedGenCmd = &cli.Command{
+	Name: "debug-feed",
+	Action: func(cctx *cli.Context) error {
+		xrpcc, err := cliutil.GetXrpcClient(cctx, true)
+		if err != nil {
+			return err
+		}
+
+		didr := cliutil.GetDidResolver(cctx)
+
+		uri := cctx.Args().First()
+		puri, err := util.ParseAtUri(uri)
+		if err != nil {
+			return err
+		}
+
+		ctx := context.TODO()
+
+		out, err := atproto.RepoGetRecord(ctx, xrpcc, "", puri.Collection, puri.Did, puri.Rkey)
+		if err != nil {
+			return fmt.Errorf("getting record: %w", err)
+		}
+
+		fgr, ok := out.Value.Val.(*bsky.FeedGenerator)
+		if !ok {
+			return fmt.Errorf("invalid feedgen record")
+		}
+
+		fmt.Println("Feed DID is: ", fgr.Did)
+		doc, err := didr.GetDocument(ctx, fgr.Did)
+		if err != nil {
+			return err
+		}
+
+		fmt.Println("Got service did document:")
+		b, err := json.MarshalIndent(doc, "", "  ")
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(b))
+
+		var ss *did.Service
+		for _, s := range doc.Service {
+			if s.ID.String() == "#bsky_fg" {
+				cp := s
+				ss = &cp
+				break
+			}
+		}
+
+		if ss == nil {
+			return fmt.Errorf("No '#bsky_fg' service entry found in feedgens DID document")
+		}
+
+		fmt.Println("Service endpoint is: ", ss.ServiceEndpoint)
+
+		fgclient := &xrpc.Client{
+			Host: ss.ServiceEndpoint,
+		}
+
+		skel, err := bsky.FeedGetFeedSkeleton(ctx, fgclient, "", uri, 30)
```

Comment:
this might fail if the route is behind auth

may be worth disambiguating in the error handling - but also since this is used by devs, probably not a big deal

Reply:
Ah, good point. Is there any way for me to generate the correct auth token?